examples/hubert/dataset/hubert_dataset.py

```diff
@@ -48,26 +54,30 @@ def __init__(
             raise AssertionError("``min_len`` should be non-negative and smaller than ``max_len``")
         if max_token_count is not None and batch_size is not None:
             raise AssertionError("The ``max_token_count`` and ``batch_size`` can't be both set.")
+        if max_token_count is None and batch_size is None:
+            raise AssertionError("One of ``max_token_count`` or ``batch_size`` must be set.")
+        if max_token_count is not None:
+            assert (
+                max_len <= max_token_count
+            ), "The  ``max_token_count`` must be greater than or equal to the maximum value of ``lengths``."
         # Filter out samples which are outside the bounds of [min_len, max_len]
-        # sort to minimize gap when bucketizing.
         filtered_length_idx = [(length, i) for i, length in enumerate(lengths) if min_len <= length <= max_len]
         if len(filtered_length_idx) == 0:
             raise AssertionError("``lengths`` cannot be empty after filtering.")
-        sorted_filtered_length_idx = sorted(filtered_length_idx, key=lambda x: x[0])
-        self.lengths = [e[0] for e in sorted_filtered_length_idx]
-        self.indices = [e[1] for e in sorted_filtered_length_idx]
+        self.lengths = [e[0] for e in filtered_length_idx]
```

Comment:
I feel like we still need to sort the lengths for one time. If `shuffle=False`, then we will generate the batch from the sorted index list, to minimize the padding.